### PR TITLE
Lint markdown in CI, and fix existing lint, from sylabs pr 174, 176 & 186

### PR DESCRIPTION
**Pull request by DrDaveD**
This PR pulls in and adapts changes from sylabs PRs
- sylabs/singularity#174
- sylabs/singularity#176
- sylabs/singularity#186
which addressed issues
- sylabs/singularity#173
- sylabs/singularity#175
- sylabs/singularity#185

Note that this doesn't change any application code, only documentation.

The original PR descriptions were:

Use markdownlint in CI, and fix the existing lint.

General edits and removal of obsolete GOPATH location requirement in INSTALL.md

Ensure CoC is recognized by GitHub by placing it in expected file.

### CODE REVIEW

**Review by gmkurtzer — COMMENTED** [review]
There is a lot here, some of which I am strongly against and needs to go through proper validation and checking before adoption just because sylabs did it (e.g. the Code of Conduct).

I would review the lint fixes alone if they were separated from the rest of this.

Thanks!

**Comment by DrDaveD** [comment]
There was no change to the Code of Conduct, it was just moved to a separate file so it is recognized by Github.  If you'd like to see that change separately please click on the commit.  I thought I should include it in the same PR because it was dependent on the previous changes to CONTRIBUTING.md.

The lint fixes alone account for the vast majority.   You can see those in the first commit.  In fact for this PR I recommend looking at each of the 4 commits separately.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting Jason Tuschen (tusch@sylabs.io). All complaints will be
```
Comment (ikaneshiro) [inline CODE_OF_CONDUCT.md]: This contact should corrected

Comment (DrDaveD) [reply CODE_OF_CONDUCT.md]: Good catch.  I don't know how I missed that.  I remember seeing that difference.  Fixed.

File: COPYRIGHT.md [hunk 2]
```diff
@@ -1,4 +1,6 @@
-Copyright (c) 2017-2018, Sylabs, Inc. All rights reserved.
+# Copyright
+
+Copyright (c) 2017-2021, Sylabs, Inc. All rights reserved.
```
Comment (gmkurtzer) [inline COPYRIGHT.md]: This is the project's overarching copyright. The project is not copyrighted to Sylabs, but Sylabs does have copyright assertions on any code/sources that we cherry pick so this is unecessary.

As a matter of fact, this should actually be 2019-2021 HPCng.

Comment (DrDaveD) [reply COPYRIGHT.md]: Sure, but we have a separate issue for updating all the copyright messages.  Can we defer it until then?  Meanwhile I will remove the change in this file.

**Review by ikaneshiro — APPROVED** [review]

**Pull request merged**